lib/ui/painting/image_decoder_impeller.cc

```diff
@@ -446,24 +461,39 @@ void ImageDecoderImpeller::Decode(fml::RefPtr<ImageDescriptor> descriptor,
           result(nullptr, bitmap_result.decode_error);
           return;
         }
-        auto upload_texture_and_invoke_result = [result, context,
-                                                 bitmap_result]() {
-          // TODO(jonahwilliams): remove ifdef once blit from buffer to
-          // texture is implemented on other platforms.
+        auto upload_texture_and_invoke_result = [result, context, bitmap_result,
+                                                 gpu_disabled_switch]() {
+          // TODO(jonahwilliams): remove ifdef once blit from buffer
+          // to texture is implemented on other platforms.
           sk_sp<DlImage> image;
           std::string decode_error;
+          bool needs_upload = true;
+
 #ifdef FML_OS_IOS
-          std::tie(image, decode_error) = UploadTextureToPrivate(
-              context, bitmap_result.device_buffer, bitmap_result.image_info);
-#else
-          std::tie(image, decode_error) =
-              UploadTextureToShared(context, bitmap_result.sk_bitmap);
-#endif
+          FML_DCHECK(gpu_disabled_switch);
+          if (gpu_disabled_switch) {
+            gpu_disabled_switch->Execute(fml::SyncSwitch::Handlers().SetIfFalse(
```

Comment:
It's worth adding a comment about the omission of a true branch.

Reply:
This got moved and there's no longer an omission.